src/frontend/chat_room/reducer/index.ts

```diff
@@ -0,0 +1,23 @@
+import {MessageStatus, SingleMessage} from "../components/ChatRoomBubbles";
+import {combineReducers} from "redux";
+import update from 'react-addons-update';
+import {ChatRoomActions} from "../Actions";
+
+export interface ChatRoomState {
+    messages: Array<SingleMessage>;
```

Comment:
I wonder if you need to keep track of which user is sending which message.

Reply:
We only care about `Received` and `Sent` to render the messages. I think I can add one more field `Sender`.